crypto/provider.c

```diff
@@ -54,7 +56,19 @@ const OSSL_PARAM *OSSL_PROVIDER_gettable_params(const OSSL_PROVIDER *prov)
 
 int OSSL_PROVIDER_get_params(const OSSL_PROVIDER *prov, OSSL_PARAM params[])
 {
-    return ossl_provider_get_params(prov, params);
+    OSSL_PARAM *p;
+    int ret;
+    unsigned int status = 0;
+
+    ret = ossl_provider_get_params(prov, params);
+
+    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_SELF_TEST);
+    if (p != NULL) {
+        OSSL_PARAM_get_uint(p, &status);
+        if (status == 0)
+            evp_method_store_flush(ossl_provider_library_context(prov));
```

Comment:
Is this flushing good enough? It doesn't actually prevent the FIPS module being used if algorithms have been pre-fetched - which is a very reasonable strategy and is something libssl does. In the libssl case I could imagine that you could get a self test failure and libssl would carry on quite happily still using the FIPS module.

Reply:
This was discussed in the initial design meetings, Not sure there is much we can do if they are prefetched..
The old FOm had a test (or at least tried to and failed) in every operation to see. Not sure that we want to head down that route do we?

Reply:
@paulidale - care to comment?

Reply:
@mattcaswell - I think this is outside the scope of this PR. A separate PR will be needed to address this.

Reply:
Ok